powershell/public/maester/azure/Get-AzureRootManagementGroup.ps1

```diff
@@ -0,0 +1,42 @@
+﻿<#
+.SYNOPSIS
+    Gets the Azure Tenant Root Management Group ID
+
+.DESCRIPTION
+    This function retrieves the Tenant Root Management Group ID by querying all management groups
+    and finding the one with the display name "Tenant Root Group".
+
+.EXAMPLE
+    Get-AzureRootManagementGroup
+
+    Returns the ID of the Tenant Root Management Group if found, otherwise returns $null.
+
+.LINK
+    https://maester.dev/docs/commands/Get-AzureRootManagementGroup
+#>
+function Get-AzureRootManagementGroup {
+    [CmdletBinding()]
+    [OutputType([string])]
+    param()
+
+    try {
+        $mgResponse = Invoke-MtAzureRequest `
+            -RelativeUri "/providers/Microsoft.Management/managementGroups" `
+            -ApiVersion "2020-05-01"
+
+        $mgList = $mgResponse.value
+    }
+    catch {
+        Write-Verbose "Error retrieving management groups: $_"
+        return $null
+    }
+
+    $rootGroup = $mgList | Where-Object { $_.properties.displayName -eq "Tenant Root Group" }
```

Comment:
The tenant root group displayName can be changed, and I have seen advise in some places that encourages people to set their own displayName for it. What I learned in reviewing this is that the name (GUID) of the tenant root group always matches the tenant ID. Given that, it looks like the most reliable check would be to look for a management group that has a `name` attribute that matches the tenant ID or the `id` attribute matches `/providers/Microsoft.Management/managementGroups/{tenantId}`.